src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -8571,7 +8575,7 @@ private static bool CompletionRequiresQuotes(string completion, bool escape)
             {
                 requireQuote = false;
                 var value = tokens[0].Text;
-                if (value.IndexOfAny(charToCheck) != -1)
+                if (ContainsCharsToCheck(value, escape))
```

Comment:
I wonder why `if` is used instead of `requireQuote = ContainsCharsToCheck(value, escape)`

Reply:
Good spot. I think this code can be simplified to just call method.

Reply:
Fixed in 45148f9296735b53e6208e4c4fc460bc291c14ce